Startobs

dstopp opened the pull request:
Initial stab at replacing page w/startobs.

dstopp committed:
bunch of white space fix ups

dstopp committed:
first pass

dstopp committed:
Merge branch 'new_ui' of https://github.com/SETI/pds-opus into startobs

# Conflicts:
#	opus/application/static_media/js/browse.js
#	opus/application/static_media/js/cart.js
#	opus/application/static_media/js/hash.js

dstopp committed:
-removed page markers
-infinite scroll gets next obs number correctly scrolling down, up not yet tested
-slider not yet tested

dstopp committed:
Merge branch 'new_ui' of https://github.com/SETI/pds-opus into startobs

dstopp committed:
added slider support for obsnum

dstopp committed:
fixes some of the scroll interaction w/the slider

dstopp committed:
a few minor loading status changes

dstopp committed:
Merge branch 'new_ui' of https://github.com/SETI/pds-opus into startobs

dstopp committed:
Merge branch 'new_ui' of https://github.com/SETI/pds-opus into startobs

dstopp committed:
-remove unnecessary console.log
-toggle the dataTable on add to cart

dstopp committed:
intermediate infinite scroll checkin

dstopp committed:
minor change to fix startobs on slider

dstopp committed:
-countGalleryImage - make specific to which tab, browse vs. cart

dstopp committed:
a few more cleanup issues

dstopp committed:
fix for addrange and other cart issues.

dstopp committed:
Merge branch 'new_ui' of https://github.com/SETI/pds-opus into startobs

# Conflicts:
#	opus/application/static_media/js/browse.js
#	opus/application/static_media/js/cart.js
#	opus/application/static_media/js/opus.js

dstopp committed:
-initial bits seem to have infinite scroll in cart working with startobs

dstopp committed:
typo

dstopp committed:
fixed startobs issue; needed to start on last+1

dstopp committed:
Merge branch 'new_ui' of https://github.com/SETI/pds-opus into startobs

# Conflicts:
#	opus/application/static_media/js/browse.js
#	opus/application/static_media/js/cart.js

dstopp committed:
Merge branch 'new_ui' of https://github.com/SETI/pds-opus into startobs

# Conflicts:
#	opus/application/static_media/js/hash.js
#	opus/application/static_media/js/opus.js

rfrenchseti committed:
Merge branch 'startobs' of https://github.com/dstopp/pds-opus into dstopp-startobs

dstopp committed:
-fix for slider 'jump'
-change the prefetch to just get 2x limit instead of using getNextPage

dstopp committed:
-additional slider fix

dstopp committed:
fixed issue w/partial fill

dstopp committed:
-fix add/remove cart issue
-don't retrieve next page /infiniteScroll in cart
-adjustment to countGalleryImages - still needs thought.

dstopp committed:
-for debugging; added title to table element so i can see the obs num on hover
-replaced detach with remove to prevent leaks
-added reRender if block to fix sort
-minor init issue w/galleryData

dstopp committed:
-fix for double empty message

dstopp committed:
-fixed the metadata select dialog
-added shift+click anywhere on a row for table view observation selection
-disabled windows highlight of table items on shift+click

dstopp committed:
fixed the table checkbox on add of single observation

rfrenchseti commented on opus/application/static_media/js/browse.js:
```diff
@@ -51,18 +51,18 @@ var o_browse = {
+            let tab = view.namespace;
+            if (opus.prefs[`${view.prefix}startobs`] > 0) {
+                let prev = $(`${view.namespace} [data-obs]`).first().data("obs") - o_browse.getLimit();
+                if ($(`${tab} .gallery-contents`).scrollTop() || $(`${tab} .dataTable`).scrollTop() === 0) {
```
The use of || here is confusing me. As written, this says "If gallery-contents scrollTop is NOT zero, OR dataTable scrollTop IS zero". Don't you want to say if either is zero?

rfrenchseti commented on opus/application/static_media/js/browse.js:
```diff
@@ -957,19 +958,18 @@ var o_browse = {
-                $(".dataTable tbody").append(tableHtml);
+            // wondering if there should be more logic here to determine if the new block of observations
+            // is contiguous w/the existing block of observations, not just before/after...
+            if ($(`${namespace} .thumbnail-container`).first().data("obs") > data.start_obs) {
```
If the gallery is empty, isn't there no first() element? In which case I assume first().data("obs") returns "undefined". That makes the if confusing. Perhaps you should check if there's a first() element at all, and only then do the comparison?

rfrenchseti committed:
Merge branch 'startobs' of https://github.com/dstopp/pds-opus into dstopp-startobs

dstopp committed:
-make the table 'tools' cell not overlap
-remove unused file gallery.html

rfrenchseti committed:
Merge branch 'startobs' of https://github.com/dstopp/pds-opus into dstopp-startobs

rfrenchseti committed:
Fix bug with CSV file when downloading single OPUSID. Fix bug with old slugs and startobs in normalizeurl. Remove old colls= slug.

rfrenchseti committed:
Snapshot of API and test cleanup.

rfrenchseti committed:
Cart return format tests.

rfrenchseti committed:
help return format tests.

rfrenchseti committed:
Clean up results APIs, add search return format tests.

rfrenchseti committed:
UI API cleanup and UI return format tests.

rfrenchseti committed:
Clean up function names.

dstopp committed:
-metadata modal updated to get next chunk of infinite scroll when at the end
-added hover text to next/prev hands
-removed unused code
-some debug code for infinite scroll

dstopp committed:
Merge pull request #4 from rfrenchseti/api_cleanup_190423

Api cleanup 190423

dstopp committed:
-undo range by clicking on ESC

dstopp committed:
Merge branch 'startobs' of https://github.com/dstopp/pds-opus into startobs

dstopp committed:
Merge branch 'new_ui' of https://github.com/SETI/pds-opus into startobs

dstopp committed:
Merge branch 'new_ui' of https://github.com/SETI/pds-opus into startobs

# Conflicts:
#	opus/application/apps/results/views.py
#	opus/application/static_media/js/browse.js
#	opus/application/static_media/js/cart.js
#	opus/application/static_media/js/opus.js
